### PR TITLE
Feature validations and model specs

### PR DESCRIPTION
**Pull request by admirerbrown**
Hello Sir 🤝,
In this feature branch i performed following requirements:

 🟢 **Project requirements**
Add the following validations:

> For the User model:
-  Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.

> For the Post model:
- Title must not be blank.
-  Title must not exceed 250 characters.
-  CommentsCounter must be an integer greater than or equal to zero.
-  LikesCounter must be an integer greater than or equal to zero.

> - Add unit specs for all of your models' methods and validations.

NB: All test cases passed. see below
Screenshot from 2023-03-30 22-00-49

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    expect(likes_counter).to be >= 0
+  end
+
+  it '#update_post_counter should increase post_counter by 1' do
+    post.update_post_counter
+    check = user1.posts_counter
+
+    expect(check).to be > 0
+  end
```
Comment (danmainah) [inline spec/models/post_spec.rb:37]: - [ ] @admirerbrown great work on your test however you forgot to add a test for `recent_comments` method kindly add test for this method

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
+    Post.create(author: first_user, title: 'Hello', text: 'This is my second post')
+    Post.create(author: first_user, title: 'Hello', text: 'This is my third post')
+    Post.create(author: first_user, title: 'Hello', text: 'This is my fourth post')
+    Post.create(author: first_user, title: 'Hello', text: 'This is my fifth post')
+    curr_posts = first_user.recent_posts
+    result = curr_posts.size
+
+    expect(result).to eq(3)
+  end
```
Comment (danmainah) [inline spec/models/user_spec.rb:36]: - [ ] @admirerbrown great work on your tests, however, you forgot to add tests for the like and comment models , kindly add tests for the models, this is a project requirement 
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @admirerbrown , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by victorokoroji — APPROVED** [review]
Hi @admirerbrown, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- All tests passed✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**